File: #0451.sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {

        unordered_map<char, int> m;
        for (char c : s) {
            m[c]++;
        }
        
        std::sort(s.begin(), s.end(),
            [&m](const char l, const char r) {
                return (m[l] == m[r]) ? l < r : m[l] > m[r];
        });
        return s;
    }
};

File: #0451.sort-characters-by-frequency.js
/**
 * @param {string} s
 * @return {string}
 */
var frequencySort = function(s) {
    let map = {};
    for(var i = 0; i < s.length; i++) {
        map[s[i]] = map[s[i]] ? map[s[i]] + 1 : 1;
    }

    var keys = Object.keys(map).sort((left, right) => {
        return map[right] - map[left];
    });

    return "".concat(...keys.map((item) => {
        return item.repeat(map[item]);
    }));
};
